src/Simulator/IncrementalRedstoneSimulator/RedstoneComparatorHandler.h

```diff
@@ -54,7 +54,13 @@ class cRedstoneComparatorHandler : public cRedstoneHandler
 
 			virtual bool Item(cBlockEntity * a_BlockEntity) override
 			{
-				auto & Contents = static_cast<cBlockEntityWithItems *>(a_BlockEntity)->GetContents();
+				auto BlockEntityWithItems = dynamic_cast<cBlockEntityWithItems *>(a_BlockEntity);
+				if (!BlockEntityWithItems)
```

Comment:
We use explicit comparison to `nullptr` for pointers. `!var` is only ever used for bool variables. Please change this.